### DISCUSSION TITLE
Missing Documentation on how to add and/or customize existing language Dockerfiles

### DISCUSSION THREAD

**Discussion by conradwt** [microsoft/vscode-docker#3247] [Q&A] [answered]
_Originally posted by @bwateratmsft in https://github.com/microsoft/vscode-docker/issues/2026#issuecomment-635974657_

However, I cannot find any reference to the documentation on how-to add and/or customize the generated Dockerfile using VSCode.

**Comment by bwateratmsft** [comment, marked as answer]
Converted to a discussion.

The extension uses [Handlebars](https://handlebarsjs.com/) templates to scaffold Dockerfiles (and other Docker-related assets). By default it uses the templates within the extension install directory, in `resources/templates`, but the path for templates can be changed using the `docker.scaffolding.templatePath` setting in VSCode. Any templates you place within that folder specified by the setting will be used; if absent then the templates from the extension install path will be used. You would want to use the same folder structure as what is under the installed templates.

**Reply by conradwt** [reply]
Will take a look this weekend and report back early next week.
